### PR TITLE
Add status (phase and provider id) to get output

### PR DESCRIPTION
**Pull request by jichenjc**
Add output to get command about the status of machine and provider ID as well

part of bug #820

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: config/rbac/rbac_role.yaml [hunk 1]
```diff
@@ -89,7 +89,6 @@ rules:
   - cluster.k8s.io
   resources:
   - machines
-  - machines/status
```
Comment (detiber) [inline config/rbac/rbac_role.yaml]: This change looks unintentional. Are we missing an rbac annotation for this?

File: config/crds/cluster_v1alpha1_machine.yaml [hunk 2]
```diff
@@ -6,6 +6,15 @@ metadata:
     controller-tools.k8s.io: "1.0"
   name: machines.cluster.k8s.io
 spec:
+  additionalPrinterColumns:
+  - JSONPath: .spec.providerID
+    description: Provider ID
+    name: ProviderID
+    type: string
+  - JSONPath: .status.phase
+    description: status of machine
```
Comment (vincepri) [inline config/crds/cluster_v1alpha1_machine.yaml]:
```suggestion
    description: Phase represents the current machine status (e.g. Pending, Running, Terminating, Failed, etc.).
```

Comment (jichenjc) [reply config/crds/cluster_v1alpha1_machine.yaml]: ok, will do

File: config/rbac/rbac_role.yaml (hunk 1 above)
Comment (jichenjc) [reply config/rbac/rbac_role.yaml]: looks like I need another PR..

File: config/crds/cluster_v1alpha1_machine.yaml (hunk 2 above)
Comment (detiber) [reply config/crds/cluster_v1alpha1_machine.yaml]: It looks like this comment still applies? Maybe a result of the update to remove the RBAC change?

**Comment by jichenjc** [comment]
please note comma ,   is not allowed in printcolumn, so I use / instead

**Pull request merged**